File: lua/yn.lua
#!/usr/bin/env lua

local main
main = function()
  local parser="http://yubnub.org/parser/parse?command="

  local command=table.concat(arg, "+")

  os.execute("open " .. parser .. command)
end

if type(package.loaded[(...)]) ~= "userdata" then
  main(arg)
else
  module(..., package.seeall)
end

File: ruby/yn.rb
#!/usr/bin/env ruby

# Author:: Andrew Pennebaker (andrew.pennebaker@gmail.com)
#
# KOPIMI
#
# == Usage
#
# yn <command>

def main
  parser = 'http://yubnub.org/parser/parse?command='

  command = ARGV.join '+'

  system "open #{parser}#{command}"
end

main if $PROGRAM_NAME == __FILE__
